src/XrdHttp/XrdHttpSecurity.cc

```diff
@@ -164,6 +165,7 @@ XrdHttpProtocol::HandleGridMap(XrdLink* lp)
       TRACEI(DEBUG, " Mapping name: '" << SecEntity.moninfo << "' --> " << bufname);
       if (SecEntity.name) free(SecEntity.name);
       SecEntity.name = strdup(bufname);
+      SecEntity.eaAPI->Add("gridmap.name", bufname, true);
```

Comment:
I see this consistently set but who is actually going to look at that? I thought the whole idea is that the entity.name would hold this value. So, I must be mistaken. Can you explain the reasoning here (I am sure it will be obvious once you do).

Reply:
It's not obvious!

It's a bit of a problem of order of precedence.  For the final value of `entity.name`, want the following order:
1.  Explicit entries in the grid mapfile.
2. Explicit entries in the VOMS mapfile
3. Whatever the default behavior is for the GSI plugin as a whole (e.g., potentially an auto-generated name).

Unfortunately, the naive solution results in an ordering of (1), (3), then (2).  Thus, we add any explicit usage of the grid mapfile to the entity so we can distinguish a positive entry from a default fallback.

Reply:
The VOMS plugin runs after the GSI authentication logic; the GSI authentication logic that generates the `entity.name` may be based on the mapfile _or_ the default logic.  What's added here allows one to differentiate those two cases.

We need to know the difference between the default and explicit mapping because, in one case (the default), the VOMS mapfile overrides the `entity.name` value.  In the other case, the VOMS mapfile should keep the `entity.name` untouched.

Reply:
OK, I now get it. However, couldn't we accomplish the same thing with a one bit flags to indicate the type of ID in the the name. Then Voms could do the right thing. That would be more straightforward than recording the full name as it looks like it going to be used for some other purpose that a simple "type" check. BTW we do have bit in the secentity structure. What do you think?